ext/ffi/lib.rs

```diff
@@ -1244,6 +1343,19 @@ where
           pointer: cif.call::<*const u8>(*fun_ptr, &call_args),
         }
       }
+      NativeType::Struct(fields) => {
+        let size = NativeType::compute_struct_size(fields);
+        let mut result = vec![0u8; size].into_boxed_slice();
```

Comment:
I would definitely allocate the return value `Uint8Array` in JavaScript, pass it in as an extra argument "on top" and write into it here. That will need to be done when we implement Fast API support anyway.

Reply:
For that we'd have to implement size calculation of struct in JS land too right? And with padding that is.

Reply:
What about keeping it as is for first pass? When Fast API support gets added, it can be changed.

Reply:
> For that we'd have to implement size calculation of struct in JS land too right? And with padding that is.

This could call into Rust.
Or sure, it could easily be done in JS

Reply:
Calling into Rust will be more overhead than JS impl itself, I think. Fast ops won't work for passing JS objects (here, the struct definition). Well, if this is really needed right now, can do.

Reply:
Yeah, I would do this as an op despite the overhead. The result can be cached anyway.